pkg/tenant/tenant_test.go

```diff
@@ -111,6 +111,11 @@ func (suite *tenantTestSuite) SetupSuite() {
 	assert.NotEmpty(suite.srv.URL(), "no url to connect the client on")
 	suite.client, err = api.New(suite.srv.URL())
 	assert.NoError(err, "could not initialize the Tenant client")
+
+	// Set the Ensign client on the server
+	ensignClient, err := suite.ensign.Client(context.Background())
+	assert.NoError(err, "could not initialize the Ensign mock client")
+	suite.srv.SetEnsignClient(ensignClient)
```

Comment:
This is the piece we were missing in the test suite and I think the reason we've been seeing the internal server errors coming back when we try to test with the ensign mock. This piece of code just directly injects the mock client into the Tenant service so that when we're running under the test suite, the client actually points to the Ensign server mock.